flag_engine/identities/models.py

```diff
@@ -32,22 +50,54 @@ class IdentityModel:
     def composite_key(self):
         return self.generate_composite_key(self.environment_api_key, self.identifier)
 
-    def get_all_feature_states(
-        self, environment: EnvironmentModel
-    ) -> typing.List[FeatureStateModel]:
-        all_feature_states = {fs.feature: fs for fs in environment.feature_states}
+    def _get_env_feature_states_dict(
+        self,
+        environment: EnvironmentModel,
+        feature_name: str = None,
+    ) -> typing.Dict[FeatureModel, FeatureStateModel]:
+        feature_states = environment.feature_states
+        if feature_name:
+            feature_states = [environment.get_feature_state(feature_name)]
+        return {fs.feature: fs for fs in feature_states}
+
+    def _update_feature_state_dict_with_segment_override(
```

Comment:
As per the comment below regarding mutation, I think we should just have this as a `_get_matching_segment_feature_states` or something?